### PR TITLE
Feature/drawer component

### PR DESCRIPTION
**Pull request by rutgerbakker95**
- [x] I updated the changelog
- [x] I updated the docs
- [x] I wrote tests
- [x] I am proud of my code

### CODE REVIEW

File: docs/components/drawer.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+<ButtonBase class="margin-y" label="Open Drawer" @click="isOpen = true" />
+
+<Drawer
+  title="Example Drawer"
+  :show="isOpen"
+  @drawer:close="isOpen = false"
+>
+  <p style="color: #000;">Drawer content goes here</p>
+</Drawer>
```
Comment (joostwaaijer) [inline docs/components/drawer.md:45]:
```suggestion

```
<ButtonBase class="margin-y" label="Open Drawer" @click="isOpen = true" />

<Drawer
  title="Example Drawer"
  :show="isOpen"
  @drawer:close="isOpen = false"
>
  <p style="color: #000;">Drawer content goes here</p>
</Drawer>
```

File: docs/components/drawer.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+## Props
+
+| Prop          | Type      | Default     | Description                                                    |
+|---------------|-----------|-------------|----------------------------------------------------------------|
+| as            | `String`  | `'form'`    | Element type to render (`form` or `div`)                       |
+| id            | `String`  | Required    | Unique identifier for the drawer                               |
+| title         | `String`  | Required    | Title displayed in the drawer header                           |
+| show          | `Boolean` | Required    | Controls drawer visibility                                     |
```
Comment (joostwaaijer) [inline docs/components/drawer.md:54]: Booleans don't need required. If they don't get added, they default to false.

File: src/components/Drawer.vue [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+// Props
+const props = defineProps({
+  as: {
+    type: String,
+    default: 'form',
+  },
+  id: {
+    type: String,
+    default: 'drawer',
+    required: true,
```
Comment (joostwaaijer) [inline src/components/Drawer.vue:146]: This is contradicting

File: src/tailwind/components/drawer.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+  // Filter drawer theme object
+  const drawerRootTheme = Object.entries(theme('components.drawer') || {})
+    .filter(([key]) => !reservedKeys.includes(key))
+    .reduce((acc, [key, value]) => ({ ...acc, [key]: value }), {});
+
+  return {
+    '.drawer': {
+      position: 'relative',
+      zIndex: 500,
```
Comment (joostwaaijer) [inline src/tailwind/components/drawer.js:20]: zIndexes moet je wel goed over nadenken, vooral ook omdat elk component weer in een drawer getoond moet worden. Ik heb dit in het verleden wel eens mis zien gaan met comboboxes, datepickers, etc.

**Review by joostwaaijer — CHANGES_REQUESTED** [review]

**Review by joostwaaijer — APPROVED** [review]

File: docs/components/drawer.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
+<script setup>
+import { ref } from 'vue';
+
+import { ButtonBase } from '@robuust-digital/vue-components';
+import { Drawer } from '@robuust-digital/vue-components';
```
Comment (boboldehampsink) [inline docs/components/drawer.md:5]:
```suggestion
import { ButtonBase, Drawer } from '@robuust-digital/vue-components';
```

File: tests/unit/index.js [hunk 6]
```diff
@@ -10,6 +10,17 @@ class IntersectionObserver {
...
+
+class ResizeObserver {
+  observe() { }
+
+  unobserve() { }
+
+  disconnect() { }
+}
+
+window.ResizeObserver = ResizeObserver;
```
Comment (boboldehampsink) [inline tests/unit/index.js:22]: Consider creating one Observer class and issuing that to both intersection- and resize observer

**Review by boboldehampsink — CHANGES_REQUESTED** [review]

**Pull request merged**